core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -66,7 +100,8 @@ private[spark] object TaskDescription {
     }
   }
 
-  def encode(taskDescription: TaskDescription): ByteBuffer = {
+  @throws[TaskNotSerializableException]
+  def encode(taskDescription: TaskDescription, serializedTask: ByteBuffer): ByteBuffer = {
```

Comment:
Rather than pass in serializedTask here, can this function handle the task serialization (so just accept taskDescription as an argument, and then call taskDescription.serializeTask on line 128?)